platform/admin/core/admin-core-appservice/src/main/java/org/codice/ddf/admin/application/service/impl/ApplicationServiceBean.java

```diff
@@ -519,12 +521,13 @@ public void setApplicationPlugins(List<ApplicationPlugin> applicationPlugins) {
    * @return the service or <code>null</code> if missing.
    */
   final MetaTypeService getMetaTypeService() {
-    if (serviceTracker == null) {
-      BundleContext context = getContext();
-      serviceTracker = new ServiceTracker<Object, Object>(context, META_TYPE_NAME, null);
+    BundleContext context = getContext();
+
+    if (serviceTracker == null && context != null) {
+      serviceTracker = new ServiceTracker<>(context, META_TYPE_NAME, null);
       serviceTracker.open();
     }
-    return (MetaTypeService) serviceTracker.getService();
+    return (serviceTracker != null) ? (MetaTypeService) serviceTracker.getService() : null;
```

Comment:
can we just do `return (MetaTypeService) serviceTracker.getService();` in the `if` statement above and return `null` here?

Reply:
@clockard the if statement only executes if serviceTracker is null *and* the context is not null, so we need a place to check serviceTracker on its own.